meshio/ugrid/_ugrid.py

```diff
@@ -0,0 +1,273 @@
+"""
+I/O for AFLR's UGRID format
+[1] <http://www.simcenter.msstate.edu/software/downloads/doc/ug_io/3d_grid_file_type_ugrid.html>.
+Check out
+[2] <http://www.simcenter.msstate.edu/software/downloads/ug_io/index_simsys_web.php?path=release>
+for UG_IO C code able to read and convert UGRID files
+Node ordering described in
+[3] http://www.simcenter.msstate.edu/software/downloads/doc/ug_io/3d_input_output_grids.html
+"""
+import logging
+
+import numpy
+
+from .._exceptions import ReadError
+from .._files import open_file
+from .._helpers import register
+from .._mesh import Mesh
+
+# Float size and endianess are recorded by these suffixes
+# binary files come in C-type or FORTRAN type
+# http://www.simcenter.msstate.edu/software/downloads/doc/ug_io/ugc_file_formats.html
+#
+# 64-bit versions described here
+# http://www.simcenter.msstate.edu/software/downloads/doc/ug_io/ugc_l_file_formats.html
+file_types = {
+    "ascii": {"type": "ascii", "float_type": "f", "int_type": "i"},
+    "b8l": {"type": "C", "float_type": ">f8", "int_type": ">i8"},
+    "b8": {"type": "C", "float_type": ">f8", "int_type": ">i4"},
+    "b4": {"type": "C", "float_type": ">f4", "int_type": ">i4"},
+    "lb8l": {"type": "C", "float_type": "<f8", "int_type": "<i8"},
+    "lb8": {"type": "C", "float_type": "<f8", "int_type": "<i4"},
+    "lb4": {"type": "C", "float_type": "<f4", "int_type": "<i4"},
+    "r8": {"type": "F", "float_type": ">f8", "int_type": ">i4"},
+    "r4": {"type": "F", "float_type": ">f4", "int_type": ">i4"},
+    "lr8": {"type": "F", "float_type": "<f8", "int_type": "<i4"},
+    "lr4": {"type": "F", "float_type": "<f4", "int_type": "<i4"},
+}
+
+file_type = None
+
+
+def determine_file_type(filename):
+    file_type = file_types["ascii"]
+    filename_parts = filename.split(".")
+    if len(filename_parts) > 1:
+        type_suffix = filename_parts[-2]
+        if type_suffix in file_types.keys():
+            file_type = file_types[type_suffix]
+
+    return file_type
+
+
+def read(filename):
+
+    global file_type
+
+    file_type = determine_file_type(filename)
+
+    with open_file(filename) as f:
+        mesh = read_buffer(f)
+    return mesh
+
+
+def _read_section(f, count, dtype):
+    if file_type["type"] == "ascii":
+        return numpy.fromfile(f, count=count, dtype=dtype, sep=" ")
+    else:
```

Comment:
This `else` isn't necessary.

Reply:
I am not sure I understand this.  Could you elaborate on it ? I just followed the vtk writer/reader
from here https://github.com/nschloe/meshio/blob/049c403eefbf339a2b8e8f7474c4f9210ea67f00/meshio/vtk/_vtk.py#L349

Reply:
Just remove else and unindent the block.

Reply:
Yep done in the latest commit